test/org/usfirst/frc/team2175/properties/JoystickPropertiesTest.java

```diff
@@ -7,153 +7,87 @@
-    }
-
-    @Test
-    public void testJoystickPropertyValues_AboveZero_Competition()
-            throws IllegalArgumentException, IllegalAccessException {
-        final String propertyFileDirectory = PROPERTY_FILE_DIR_SRC_COMPETITION;
-        commonTestPositiveProperties(propertyFileDirectory);
-    }
-
-    @Test
-    public void testJoystickPropertyValues_AboveZero_Practice()
-            throws IllegalArgumentException, IllegalAccessException {
-        final String propertyFileDirectory = PROPERTY_FILE_DIR_SRC_PRACTICE;
-        commonTestPositiveProperties(propertyFileDirectory);
-    }
-
-    protected void assertJoystickNameCorrect(final Object sut)
-            throws IllegalArgumentException, IllegalAccessException {
-        HashMap<String, ButtonInfo> hashMap =
+        final HashMap<String, ButtonInfo> hashMap =
                 getFieldsOfType(ButtonInfo.class, sut);
 
-        for (Map.Entry<String, ButtonInfo> entry : hashMap.entrySet()) {
-            String key = entry.getKey();
-            ButtonInfo value = entry.getValue();
-            String joystickName = value.joystickName.trim();
-            final String assertMessage =
-                    "'" + joystickName + "' in ButtonInfo field " + key
-                            + " is not a valid Joystick name.";
-            assertTrue(assertMessage,
-                    joystickName.equals("left") || joystickName.equals("right")
-                            || joystickName.equals("gamepad"));
-        }
-    }
+        for (final Map.Entry<String, ButtonInfo> entry : hashMap.entrySet()) {
+            final String key = entry.getKey();
+            final ButtonInfo value = entry.getValue();
+            final String joystickName = value.joystickName.trim();
+            final int buttonValue = value.buttonNumber;
 
-    protected void assertButtonNumbersGreaterThanZero(final Object sut)
-            throws IllegalArgumentException, IllegalAccessException {
-        HashMap<String, ButtonInfo> hashMap =
-                getFieldsOfType(ButtonInfo.class, sut);
-
-        for (Map.Entry<String, ButtonInfo> entry : hashMap.entrySet()) {
-            String key = entry.getKey();
-            ButtonInfo value = entry.getValue();
-            final String assertMessage =
-                    "ButtonInfo field " + key + " was not greater than 0.";
-            assertTrue(assertMessage, value.buttonNumber > 0);
+            checkJoystickNames(joystickName, key);
+            checkButtonNumbersAboveZero(buttonValue, key);
         }
     }
 
-    protected void commonTestJoystickNames(final String propertyFileDirectory)
-            throws IllegalArgumentException, IllegalAccessException {
-        BaseProperties.setPropertyFileDir(propertyFileDirectory);
-        final BaseProperties baseProperties = new JoystickProperties();
-
-        assertJoystickNameCorrect(baseProperties);
+    protected void checkJoystickNames(final String joystickName,
```

Comment:
@4thwind Nitpick: This method only checks a single joystick name, so maybe `checkJoystickName` would be a better name.